### PR TITLE
Makefile: improve documentation

### PR DESCRIPTION
**Pull request by kemenaran**
Add comments to make the build process easier to understand.

### CODE REVIEW

**Review by zelosos — APPROVED** [review]

**Comment by zelosos** [comment]
Is more understandable.
I would add a comment for: Are all those files converted every time or only if they changed?

File: Makefile [hunk 1]
```diff
@@ -38,20 +38,27 @@ asm_files = $(shell find src     -type f -name '*.asm')
...
+# Compile all dependencies (ASM, 2BPP) into an single object file.
+# (This means all the source code is always fully recompiled: for now,
+# we don't compile the different ASM files separately.)
+# Locale-specific rules below (e.g. `src/main.azlj.o`) will add their own
+# pre-requisites to the ones defined by this rule.
 src/main.%.o: src/main.asm $(asm_files) $(gfx_files:.png=.2bpp)
 	$(ASM) $(ASFLAGS) $($*_ASFLAGS) -i src/ -o $@ $<
 
-# Link an object file to a GBC file
+# Link an object file to a GBC file.
```
Comment (marijnvdwerf) [inline Makefile:59]:
```suggestion
# Link object files into a GBC executable rom
```

**Comment by kemenaran** [comment]
>  I would add a comment for: Are all those files converted every time or only if they changed?

Good question. That said, the very purpose of a Makefile is to recompile changed files, and _only_ changed files. So that would be like explaining what a Makefile is used for; which is surely useful, but  it probably belongs more to the documentation or the Wiki.

**Comment by kemenaran** [comment]
Thanks for the review 🙇

**Pull request merged**